src/language/templates/visitors/lookup_visitor.hpp

```diff
@@ -22,39 +22,51 @@ namespace nmodl {
 namespace visitor {
 
 /**
- * @addtogroup visitor_classes
- * @{
+ * \addtogroup visitor_classes
+ * \{
  */
 
 /**
  * \class AstLookupVisitor
  * \brief %Visitor to find AST nodes based on their types
  */
-class AstLookupVisitor: public Visitor {
+template <typename DefaultVisitor>
+class MetaAstLookupVisitor: public DefaultVisitor {
+    static const bool is_const_visitor = std::is_same<ConstVisitor, DefaultVisitor>::value;
+  using ast_t = typename std::conditional<is_const_visitor, const ast::Ast, ast::Ast>::type;
```

Comment:
It would be `std::add_const<ast::Ast>::type`. In this case, we know that we want a const type, so there is no need to invoke a type traits to do it.
IMO `std::add_const` is useful when you want to pass a type operation in another location of the program.

But in this case, we can reuse `visit_arg_traits` template type to define `ast_t`, I will update this.